iree/compiler/Dialect/HAL/IR/HALOps.cpp

```diff
@@ -1477,8 +1477,11 @@ bool InterfaceOp::isEquivalentTo(InterfaceOp other) {
   return push_constantsAttr() == other.push_constantsAttr() &&
          bindings.size() == otherBindings.size() &&
          llvm::all_of(llvm::zip(bindings, otherBindings), [](auto bindings) {
-           return OperationEquivalence::isEquivalentTo(std::get<0>(bindings),
-                                                       std::get<1>(bindings));
+           return OperationEquivalence::isEquivalentTo(
+               std::get<0>(bindings), std::get<1>(bindings),
+               OperationEquivalence::exactValueMatch,
+               OperationEquivalence::exactValueMatch,
+               OperationEquivalence::Flags::IgnoreLocations);
```

Comment:
bert still deduplicates from 896 flow.exectutable ops down to 27 with this change (which doesn't use this code)

this code is used for executable linking, and it still has correct behavior I think: 
![image](https://user-images.githubusercontent.com/4010439/128233730-e0392948-d20d-411a-b8f7-2e48540e4e39.png)

left: `google`, right: `main`

Reply:
actually, executables changed a bit (896 -> 27 on `google` compared to 1260 -> 32 on `main` before)

![image](https://user-images.githubusercontent.com/4010439/128234357-30cc1812-7a10-477f-9221-ce446bcccfd0.png)

I think that's fine? Probably compatible upstream changes. Wonder what our benchmarks will report.